### PR TITLE
feat: selectively generate libraries

### PR DESCRIPTION
**Pull request by JoeWang1127**
In this PR:
- Allow selectively generate libraries by adding `--library-names` to CLI.

We have a request of partially releasing google-cloud-java (b/331628538), which requires only generating a selective list of libraries.

This change allows a user-defined list of libraries to be generated even if the corresponding library configs are the same.

### CODE REVIEW

**Comment by blakeli0** [comment]
Compare to selectively release libraries, I think this is not a hard blocker? Because we can comment out other libraries in the branch as a workaround?

**Comment by JoeWang1127** [comment]
> Compare to selectively release libraries, I think this is not a hard blocker? Because we can comment out other libraries in the branch as a workaround?

Yes, it's not a blocker but it requires a error-prone manual step for the release manager. Besides, it's not complex to implement since we've already had `get_target_libraries` to filter library names. We only need to pass a list from CLI.

**Comment by JoeWang1127** [comment]
Integration test failed:
```
Difference in README.md:
--- 

+++ 

@@ -40,6 +40,7 @@

 

 If you are using Maven without the BOM, add this to your dependencies:

 

+<!-- {x-version-update-start:google-cloud-alloydb:released} -->

 

 ```xml

 <dependency>

@@ -60,6 +61,7 @@

 ```Scala

 libraryDependencies += "com.google.cloud" % "google-cloud-alloydb" % "0.35.0"

 ```

+
```

We need to merge #3289 first.

File: library_generation/cli/entry_point.py [hunk 1]
```diff
@@ -164,12 +185,17 @@ def __generate_repo_and_pr_description_impl(
...
+    # Pass None if we want to fully generate the repository.
     target_library_names = (
         config_change.get_changed_libraries()
         if not _needs_full_repo_generation(config_change=config_change)
         else None
     )
+    # Generate the union of two library lists if neither of the library list
+    # indicates a full generation.
+    # The deduplication is done in `generate_from_yaml`.
+    if (target_library_names is not None) and (include_libraries is not None):
```
Comment (blakeli0) [inline library_generation/cli/entry_point.py:197]: I feel this making the logic more complicated, if `include_libraries` is provided, it should just override `target_library_names`. Especially we plan to remove  `target_library_names` in the future.

File: library_generation/cli/entry_point.py [hunk 2]
```diff
@@ -191,6 +217,12 @@ def _needs_full_repo_generation(config_change: ConfigChange) -> bool:
     return not current_config.is_monorepo() or current_config.contains_common_protos()
 
 
+def _parse_library_name_from(includes: str) -> Optional[list[str]]:
+    if includes is None:
+        return None
+    return includes.split(",")
```
Comment (blakeli0) [inline library_generation/cli/entry_point.py:223]: Maybe trim the library name?

File: library_generation/cli/entry_point.py [hunk 3]
```diff
@@ -51,6 +51,15 @@ def main(ctx):
     metadata about library generation.
     """,
 )
+@click.option(
+    "--includes",
```
Comment (blakeli0) [inline library_generation/cli/entry_point.py:55]: `includes` is a little ambiguous to me that the user may not know what is the info we are including. For example, is this `api_shortname` or `library_name` or something else?  Just `library_names` or `target_library_names`(we are already using it in scripts) sounds more reasonable to me.

File: library_generation/cli/entry_point.py [hunk 4]
```diff
@@ -51,6 +51,15 @@ def main(ctx):
...
+    type=str,
+    default=None,
+    show_default=True,
+    help="""
+    A list of library names that will be generated, separated by comma.
```
Comment (blakeli0) [inline library_generation/cli/entry_point.py:60]: Can we add more docs here? I think it is basically `library_name != null ? library_name : api_shortname` from `generation_config.yaml`?

File: library_generation/cli/entry_point.py (hunk 3 above)
Comment (JoeWang1127) [reply library_generation/cli/entry_point.py]: Changed to `library_names`.

File: library_generation/cli/entry_point.py (hunk 1 above)
Comment (JoeWang1127) [reply library_generation/cli/entry_point.py]: Yes, I also found it was a bit of confusing in this scenario.

I changed this to `library_names` takes preference if specified.

File: library_generation/cli/entry_point.py (hunk 4 above)
Comment (JoeWang1127) [reply library_generation/cli/entry_point.py:60]: Changed the comment.

File: library_generation/cli/entry_point.py (hunk 2 above)
Comment (JoeWang1127) [reply library_generation/cli/entry_point.py]: Done.

File: library_generation/cli/entry_point.py [hunk 5]
```diff
@@ -191,6 +221,12 @@ def _needs_full_repo_generation(config_change: ConfigChange) -> bool:
     return not current_config.is_monorepo() or current_config.contains_common_protos()
 
 
+def _parse_library_name_from(includes: str) -> Optional[list[str]]:
+    if includes is None:
+        return None
+    return includes.strip().split(",")
```
Comment (blakeli0) [inline library_generation/cli/entry_point.py:227]: Actually I meant trim after split.

**Review by blakeli0 — APPROVED** [review]

File: library_generation/cli/entry_point.py (hunk 5 above)
Comment (JoeWang1127) [reply library_generation/cli/entry_point.py:227]: I'll make the change before merging.

**Pull request merged**